airflow/kubernetes/pod_launcher.py

```diff
@@ -14,283 +15,15 @@
-    def read_pod_events(self, pod):
-        """Reads events from the POD"""
-        try:
-            return self._client.list_namespaced_event(
-                namespace=pod.metadata.namespace, field_selector=f"involvedObject.name={pod.metadata.name}"
-            )
-        except BaseHTTPError as e:
-            raise AirflowException(f'There was an error reading the kubernetes API: {e}')
-
-    @tenacity.retry(stop=tenacity.stop_after_attempt(3), wait=tenacity.wait_exponential(), reraise=True)
-    def read_pod(self, pod: V1Pod):
-        """Read POD information"""
-        try:
-            return self._client.read_namespaced_pod(pod.metadata.name, pod.metadata.namespace)
-        except BaseHTTPError as e:
-            raise AirflowException(f'There was an error reading the kubernetes API: {e}')
-
-    def _extract_xcom(self, pod: V1Pod):
-        resp = kubernetes_stream(
-            self._client.connect_get_namespaced_pod_exec,
-            pod.metadata.name,
-            pod.metadata.namespace,
-            container=PodDefaults.SIDECAR_CONTAINER_NAME,
-            command=['/bin/sh'],
-            stdin=True,
-            stdout=True,
-            stderr=True,
-            tty=False,
-            _preload_content=False,
-        )
-        try:
-            result = self._exec_pod_command(resp, f'cat {PodDefaults.XCOM_MOUNT_PATH}/return.json')
-            self._exec_pod_command(resp, 'kill -s SIGINT 1')
-        finally:
-            resp.close()
-        if result is None:
-            raise AirflowException(f'Failed to extract xcom from pod: {pod.metadata.name}')
-        return result
-
-    def _exec_pod_command(self, resp, command):
-        if resp.is_open():
-            self.log.info('Running command... %s\n', command)
-            resp.write_stdin(command + '\n')
-            while resp.is_open():
-                resp.update(timeout=1)
-                if resp.peek_stdout():
-                    return resp.read_stdout()
-                if resp.peek_stderr():
-                    self.log.info(resp.read_stderr())
-                    break
-        return None
-
-    def process_status(self, job_id, status):
-        """Process status information for the JOB"""
-        status = status.lower()
-        if status == PodStatus.PENDING:
-            return State.QUEUED
-        elif status == PodStatus.FAILED:
-            self.log.error('Event with job id %s Failed', job_id)
-            return State.FAILED
-        elif status == PodStatus.SUCCEEDED:
-            self.log.info('Event with job id %s Succeeded', job_id)
-            return State.SUCCESS
-        elif status == PodStatus.RUNNING:
-            return State.RUNNING
-        else:
-            self.log.error('Event: Invalid state %s on job %s', status, job_id)
-            return State.FAILED
+"""This module is deprecated. Please use `kubernetes.client.models for V1ResourceRequirements and Port."""
```

Comment:
I believe this change is breaking. If someone uses old `cncf.kubernetes` provider and new Airflow and still will want to use the deprecated  `airflow.kubernets.pod_launcher` module. This change brings an implicit dependency to the next version of `cncf.kubernetes` provider.

Any proposal how to fix it?

I see few options:

1) Leave the code for the old pod_launcher in the deprecated module (not ideal because of code duplication)

2) Add minimum `cncf.kubernetes` version requirement to `cncf.kubernetes` extra (this will not work for people who already have `airflow` installed and want to upgrade it.

3) Add `cncf.kubernetes` as "preinstalled" module for Airflow (similarly as ftp, http, imap, sqlite and add minimum version required there to the new version of provider. I like this most - imho K8S provider should be pre-installed.

WDYT? Any other options? Or maybe you do not see that as a problem?

Reply:
This is a really good point. I'd be in favour of the third option here, especially as the CNCF provider doesn't introduce many additional installation requirements (though I suppose this is subject to change if other functionality is added to the provider)

Reply:
@potiuk we can have a pod_launcher_deprecated.py file with the old code. We did something similar when upgrading the kubernetesexecutor for 2.0

Reply:
That works for me @dimberman